### PR TITLE
FuseUtilsHierarchicalDict -> NDict

### PR DESCRIPTION
**Pull request by SagiPolaczek**

### CODE REVIEW

File: fuse/utils/misc/misc.py [hunk 1]
```diff
@@ -118,9 +118,9 @@ def batch_dict_to_string(batch_dict: dict) -> str:
         :return: string representation of the batch dict
         """
         res = ''
-        all_keys = FuseUtilsHierarchicalDict.get_all_keys(batch_dict)
```
Comment (SagiPolaczek) [inline fuse/utils/misc/misc.py:121]: @mosheraboh 
In the description it says that: `:param batch_dict: might be any dict`,  but there is a use of `FuseUtilsHierarchicalDict` class method.

I think such a design is not optimal so I suggest to make sure that the dictionary is a NDict instance and use instance methods or open a new utils class for such cases.


Any thoughts? 🙂

File: fuse/dl/losses/segmentation/loss_focalLoss.py [hunk 2]
```diff
@@ -92,18 +92,18 @@ def __init__(self,
         :param resize_mode:             Resize mode- either using a max pooling kernel(default), or using PyTorch interpolation ('interpolate'/'maxpool')
         """
         super().__init__(pred_name, target_name, weight)
-        self.class_weights = class_weights
+        self.class_weights = class_weights # Should be deleted? -Sagi's TEMP note
```
Comment (SagiPolaczek) [inline fuse/dl/losses/segmentation/loss_focalLoss.py:95]: @mosheraboh 

Not related to this PR, but is this line should be deleted? `class_weights` are not being passed in the `__init__` method.. But maybe I'm missing something

File: fuse/data/datasets/caching/samples_cacher.py [hunk 3]
```diff
@@ -265,10 +265,10 @@ def _load_sample_from_cache(self, sample_id: Hashable, keys: Optional[Sequence[s
         for curr_read_dir in read_dirs:
             extension_less = os.path.join(curr_read_dir, sample_hash)
             if os.path.isfile(extension_less+'.pkl.gz'):
-                loaded_sample = load_pickle(extension_less+'.pkl.gz')
+                loaded_sample = NDict(load_pickle(extension_less+'.pkl.gz'))
                 if os.path.isfile(extension_less+'.hdf5'):
                     loaded_sample_hdf5_part = load_hdf5(extension_less+'.hdf5')
-                    loaded_sample = NDict.combine(loaded_sample, loaded_sample_hdf5_part)
+                    loaded_sample = loaded_sample.merge(loaded_sample_hdf5_part)
```
Comment (mosheraboh) [inline fuse/data/datasets/caching/samples_cacher.py:271]: change merge to be "inplace" operation

File: fuse/dl/losses/segmentation/loss_focalLoss.py (hunk 2 above)
Comment (mosheraboh) [reply fuse/dl/losses/segmentation/loss_focalLoss.py:95]: Yes, good catch. please delete it.

File: fuse/utils/ndict.py [hunk 4]
```diff
@@ -286,6 +260,23 @@ def __repr__(self):
     def __contains__(self, o: str) -> bool:
         return o == self.get_closest_key(o)
 
+    def get(self, key: str, defualt_value=None):
```
Comment (mosheraboh) [inline fuse/utils/ndict.py:263]: default

**Review by mosheraboh — APPROVED** [review]
Great work!

**Pull request merged**